lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -1613,6 +1613,96 @@ export class BrowserCacheManager extends CacheManager {
         this.setInteractionInProgress(false);
     }
 
+    /**
+     * Create request retry key to cache retry status
+     * @param correlationId
+     */
+    generateRequestRetriedKey(correlationId: string): string {
+        return `${Constants.CACHE_PREFIX}.${TemporaryCacheKeys.REQUEST_RETRY}.${correlationId}`;
+    }
+
+    /**
+     * Gets the request retry value from the cache
+     * @param correlationId
+     */
+    getRequestRetried(correlationId: string): string | null {
+        const requestRetriedKey = this.generateRequestRetriedKey(correlationId);
+        return this.getTemporaryCache(requestRetriedKey);
+    }
+
+    /**
+     * Sets the request retry value to "retried" in the cache
+     * @param correlationId
+     */
+    setRequestRetried(correlationId: string): void {
+        this.logger.trace("BrowserCacheManager.setRequestRetried called");
+        const requestRetriedKey = this.generateRequestRetriedKey(correlationId);
+        this.setTemporaryCache(requestRetriedKey, "retried", false);
+    }
+
+    /**
+     * Removes the request retry value from the cache
+     * @param correlationId
+     */
+    removeRequestRetried(correlationId: string): void {
+        const requestRetriedKey = this.generateRequestRetriedKey(correlationId);
+        this.removeTemporaryItem(requestRetriedKey);
+    }
+
+    /**
+     * Caches the redirectRequest in the cache
+     * @param redirectRequest
+     */
+    cacheRedirectRequest(redirectRequest: RedirectRequest): void {
+        this.logger.trace("BrowserCacheManager.cacheRedirectRequest called");
+        const encodedValue = base64Encode(JSON.stringify(redirectRequest));
```

Comment:
You will have to ignore `onRedirectNavigate` from the request object or this will throw - you can do that by spreading `{ onRedirectNavigate, ...restParams } = redirectRequest` and then using the rest.